src/components/SUE/report/SueReportLocation.tsx

```diff
@@ -177,6 +177,71 @@ export const SueReportLocation = ({
     return <LoadingSpinner loading />;
   }
 
+  const onMapPress = async ({
+    nativeEvent
+  }: {
+    nativeEvent: { action: string; coordinate: Location.LocationObjectCoords };
+  }) => {
+    if (nativeEvent.action !== 'marker-press' && nativeEvent.action !== 'callout-inside-press') {
+      setUpdatedRegion(false);
+      setUpdateRegionFromImage(false);
+      setSelectedPosition(nativeEvent.coordinate);
+
+      try {
+        await handleGeocode(nativeEvent.coordinate);
+      } catch (error) {
+        setSelectedPosition(undefined);
+        Alert.alert(texts.sue.report.alerts.hint, error.message);
+        throw new Error(error.message);
+      }
+    }
+  };
+
+  const onMyLocationButtonPress = async (isFullScreenMap = false) => {
+    if (!isFullScreenMap) {
+      Alert.alert(texts.sue.report.alerts.hint, texts.sue.report.alerts.myLocation, [
+        {
+          text: texts.sue.report.alerts.no
+        },
+        {
+          text: texts.sue.report.alerts.yes,
+          onPress: async () => {
+            const location = position || lastKnownPosition;
```

Comment:
couldn't this happen earlier to get the `currentPosition` (even better wording than `location` i guess)? now it happens twice in the method here and again a third time at line 260.

Reply:
Yes, that's right. I'll create this at the top of the `onMyLocationButtonPress`. 👍 I will update it as `currentPosition`.